iothub/device/samples/DeviceReconnectionSample/DeviceReconnectionSample.csproj

```diff
@@ -0,0 +1,24 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
+    <LangVersion>9.0</LangVersion>
+    <RootNamespace>Microsoft.Azure.Devices.Client.Samples</RootNamespace>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="CommandLineParser" Version="2.8.0" />
+    <!-- Working around a bug between DotNetty and .NET 5.0+ by specifying the exact version of DotNetty.Common that has the fix -->
+    <PackageReference Include="DotNetty.Common" Version="0.7.1" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(Configuration)' == 'Release'">
```

Comment:
In release, runs with the NuGet, but in Debug runs with a project reference.

This will be broken in Release until we ship a preview of v2.